ui/perfherder/compare/CompareTable.jsx

```diff
@@ -79,6 +91,58 @@ export default class CompareTable extends React.PureComponent {
     }
   };
 
+  // Sorts the specified key of the array data depending on the typeof key
+  // If something changes in columns, you just need to update sortableKeys
+  // with the corresponding type
+  // The data model of the sortable keys is:
+  // name: string
+  // originalValue: number
+  // newValue: number
+  // deltaPercentage: number
+  // confidence: number
+  // originalRuns: array
+  // newRuns: array
+  // default: string
+  sortBy = (columnName, data) => {
+    let { sortOrder, sortClass } = this.state;
+    const { sortedTestName, sortedColumnName } = this.state;
+    const { testName } = this.props;
+    const sortableKeys = {
+      name: 'string',
+      originalValue: 'number',
+      originalStddevPct: 'number',
+      newValue: 'number',
+      newStddevPct: 'number',
+      deltaPercentage: 'number',
+      confidence: 'number',
+      originalRuns: 'array',
+      newRuns: 'array',
+    };
+    const sorted = true;
+    if (sorted) {
+      // ^ sorted flag becomes true after clicking for the first time any column
+      // Toggle sort order if clicked a column header several consecutive times
+      if (sortedTestName === testName && sortedColumnName === columnName) {
+        data.sort(sortHashes(columnName, sortableKeys[columnName], !sortOrder));
+      } else {
+        // sort ascending if any column was clicked for the first time
+        sortOrder = true;
+        data.sort(sortHashes(columnName, sortableKeys[columnName], !sortOrder));
+      }
+      if (sortOrder) sortClass = 'sorted-asc ';
+      else sortClass = 'sorted-desc ';
+    } else sortClass = '';
+    this.setState({
+      sorted,
+      sortOrder: !sortOrder,
+      sortedTestName: testName,
+      sortedColumnName: columnName,
+      sortClass,
+    });
+
+    this.forceUpdate();
```

Comment:
Why do you need to do a`forceUpdate`here?

Reply:
otherwise the sorted data is not visible in the UI.